content/posts/Harsh_Rao_2025_final_report.md

```diff
@@ -0,0 +1,415 @@
+---
+title: "Desktop Application & Vue Frontend Updates: GSoC 2025 Final Report"
+date: 2025-08-31T14:04:00+05:30
+author: Harsh Rao
+type: post
+---
+
+![gsoc-cover](/images/Harsh_Rao/Final_report.jpg)
+
+**Hey Everyone 😊**,
+
+Welcome to my final blog for Google Summer of Code 2025 for the project **Desktop Application & Vue Frontend Updates**
+
+### What is the Project about🖥
+---
+
+> **[_Desktop Application & Vue Frontend Updates:_](https://summerofcode.withgoogle.com/programs/2025/projects/RLcXZOiF)
+> The project works on the Vue Simulator and aims at completing the vue-simulator. We have achieved several milestones like Authentication model, a beautiful Release Pipeline for Tauri simulator, version sync for v0 and v1, a brand new Testbench UI and many more. Let's dive into them right away !!**
+
+## Major statures that have been added and merged include:
+- Authentication model for Web and Tauri Simulator
+- Release Pipeline for Tauri Simulator
+- An improved Testbench UI/UX
+- Vue-Simulator integration with the primary codebase
+- Legacy feature sync to v0 and v1
+- Tackling major bugs - The Docker Failure
+
+---
+
+[<h3>Project Board</h3>](https://github.com/orgs/CircuitVerse/projects/43/views/2) 
+
+---
+# Authentication Model for Web and Tauri Simulator
+Let's begin with the very own Authentication model for vue-simulator that we recently published. The reason we needed a seperate Authentication model for vue simulator is because of Tauri. Earlier we were simply changing the `path` in the `URL` to direct to the Login page of Circuitverse for Authentication. This particular method fails for the Tauri simulator because:<br> 
+- It is a standalone simulator with no connection to the primary codebase.
+- We want the Tauri simulator to also work when user is offline.
+- We cannot redirect links like we do in web in Tauri.<br>
```

Comment:
Write in such way that a non technical circuitverse users can understand upcoming changes they see from this project